tdvt/tdvt/tdvt.py

```diff
@@ -381,38 +393,67 @@ def usage_text():
+                        required=False, default=None, const='', nargs='?')
+    parser.add_argument('--logp', dest='logical_pattern',
+                        help='Only run logical tests whose name and path matches the supplied string. this is a glob pattern. Also set the tds-pattern to use when running the test. Use a ? to replace the logical query config component of the test name.',  # noqa: E501
+                        required=False, default=None, const='', nargs='?')
+    parser.add_argument('--tdp', dest='tds_pattern',
+                        help='The datasource tds pattern to use when running the test. See exp and logp arguments.',
+                        required=False, default=None, const='', nargs='?')
+    parser.add_argument('--test-ex', dest='test_pattern_exclude',
+                        help='Exclude tests whose name matches the supplied string. This is a regular expression pattern. Can be used with exp and logp arguments. Also set the tds-pattern to use when running the test.',  # noqa: E501
+                        required=False, default=None, const='', nargs='?')
     parser.add_argument('--compare-sql', dest='compare_sql', action='store_true', help='Compare SQL.', required=False)
-    parser.add_argument('--nocompare-tuples', dest='nocompare_tuples', action='store_true', help='Do not compare Tuples.', required=False)
-    parser.add_argument('--diff-test', '-dd', dest='diff', help='Diff the results of the given test (ie exprtests/standard/setup.calcs_data.txt) against the expected files. Can be used with the sql and tuple options.', required=False)
+    parser.add_argument('--nocompare-tuples', dest='nocompare_tuples', action='store_true',
+                        help='Do not compare Tuples.', required=False)
+    parser.add_argument('--diff-test', '-dd', dest='diff',
+                        help='Diff the results of the given test (ie exprtests/standard/setup.calcs_data.txt) against the expected files. Can be used with the sql and tuple options.',  # noqa: E501
+                        required=False)
     parser.add_argument('-f', dest='run_file', help='Json file containing failed tests to run.', required=False)
+    parser.add_argument('--verify', dest='smoke_test',
+                        help='Verifies the connection to a data source against your cast_calcs.*.tds, as specified in the ConnectionTest section of your .ini file.',  # noqa: E501
+                        required=False, default=None, const='', nargs='?')
     return parser
 
+
 def init():
     parser = create_parser()
     args = parser.parse_args()
-    #Create logger.
-    logging.basicConfig(filename='tdvt_log_combined.txt',level=logging.DEBUG, filemode='w', format='%(asctime)s %(message)s')
+    if (args.smoke_test is not None) and (args.ds is not None):
+        args.expression_only = 'ConnectionTest'
```

Comment:
I was hoping we could use the smoke test to verify the connection before running the rest of the tests. One way could be to run the pretest against the calcs and stapes tds files then abort the rest of the tests if either of those fail. It would be good to make this flexible enough that it could support a smoke test against any arbitrary tds file. So maybe it could default to calcs and staples and then let users add more in the ini. 

I think this is still a useful first step so that users can easily verify their connections.

Reply:
Sure, sounds good! I'll keep this in place and add that functionality as well.